x/group/simulation/operations.go

```diff
@@ -0,0 +1,938 @@
+					return simtypes.NoOpMsg(group.ModuleName, group.TypeMsgVote, "voting period ended: skipping"), nil, nil
+				}
+				break
+			}
+		}
+
+		// return no-op if no proposal found
+		if proposalID == -1 {
+			return simtypes.NoOpMsg(group.ModuleName, group.TypeMsgVote, "no proposals found"), nil, nil
+		}
+
+		msg := group.MsgVoteRequest{
+			ProposalId: uint64(proposalID),
+			Voter:      acc1.Address.String(),
+			Choice:     group.Choice_CHOICE_YES,
+			Metadata:   []byte(simtypes.RandStringOfLength(r, 10)),
+		}
+		txGen := simappparams.MakeTestEncodingConfig().TxConfig
+		tx, err := helpers.GenTx(
+			txGen,
+			[]sdk.Msg{&msg},
+			fees,
+			helpers.DefaultGenTxGas,
+			chainID,
+			[]uint64{account.GetAccountNumber()},
+			[]uint64{account.GetSequence()},
+			acc1.PrivKey,
+		)
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, group.TypeMsgUpdateGroupAccountComment, "unable to generate mock tx"), nil, err
+		}
+
+		_, _, err = app.Deliver(txGen.TxEncoder(), tx)
+
+		if err != nil {
+			if strings.Contains(err.Error(), "group was modified") || strings.Contains(err.Error(), "group account was modified") {
+				return simtypes.NoOpMsg(group.ModuleName, msg.Type(), "no-op:group/group-account was modified"), nil, nil
+			}
+			return simtypes.NoOpMsg(group.ModuleName, msg.Type(), "unable to deliver tx"), nil, err
+		}
+
+		return simtypes.NewOperationMsg(&msg, true, ""), nil, err
+	}
+}
+
+// SimulateMsgExec generates a MsgExec with random values
+func SimulateMsgExec(ak exported.AccountKeeper,
+	bk exported.BankKeeper, queryClient group.QueryClient) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, sdkCtx sdk.Context, accounts []simtypes.Account, chainID string) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		acc1 := accounts[0]
+
+		account := ak.GetAccount(sdkCtx, acc1.Address)
+
+		spendableCoins := bk.SpendableCoins(sdkCtx, account.GetAddress())
+		fees, err := simtypes.RandomFees(r, sdkCtx, spendableCoins)
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, group.TypeMsgVote, "fee error"), nil, err
+		}
+		ctx := regentypes.Context{Context: sdkCtx}
+		groupAdmin, _, op, err := getGroupDetails(sdkCtx, queryClient, acc1)
+		if err != nil {
+			return op, nil, err
+		}
+		if groupAdmin == "" {
+			return op, nil, nil
+		}
+
+		result, err := queryClient.GroupAccountsByAdmin(ctx, &group.QueryGroupAccountsByAdminRequest{Admin: groupAdmin})
+		if err != nil {
+			return simtypes.NoOpMsg(group.ModuleName, group.TypeMsgVote, "fail to query group info"), nil, err
+		}
+
+		groupAccounts := result.GetGroupAccounts()
+		if len(groupAccounts) == 0 {
+			return simtypes.NoOpMsg(group.ModuleName, group.TypeMsgVote, "no group account found"), nil, nil
+		}
+
+		proposalsResult, err := queryClient.ProposalsByGroupAccount(ctx, &group.QueryProposalsByGroupAccountRequest{Address: groupAccounts[0].Address})
```

Comment:
Can we add helper for `groupAccountAddress`, Same code present in other functions.